app/controllers/buildings_controller.ts

```diff
@@ -0,0 +1,87 @@
+import Building from '#models/building'
+import Campus from '#models/campus'
+import { HttpContext } from '@adonisjs/core/http'
+export default class BuildingsController {
+  async getAll({ response }: HttpContext) {
+    try {
+      const buildings = await Building.all()
+      if (buildings.length === 0) {
+        return response.status(404).json({ message: 'No buildings in database' })
+      }
+      return response.status(200).json(buildings.map((building) => building.serialize()))
+    } catch (error) {
+      return response
+        .status(500)
+        .json({ message: 'Failed to fetch buildings', error: error.message })
+    }
+  }
+
+  async getById({ request, response }: HttpContext) {
+    try {
+      const id = request.param('id')
+      if (!id) {
+        return response.status(400).json({ message: 'Building ID is required' })
+      }
+      const building = await Building.find(id)
+      if (building) {
```

Comment:
when you are doing checks like this it's better to be specific (js logic can be wild sometimes) so i'd go with `building !== null` to explicitly show what you meant